🎉 Add docs on namespace changes (#3886)
airbytehq/airbyte: docs/understanding-airbyte/connections/README.md

When a scheduled connection is first created, a sync is executed as soon as possible. After that, a sync is run once the time since the last sync \(whether it was triggered manually or due to a schedule\) has exceeded the schedule interval. For example, consider the following illustrative scenario:

* **October 1st, 2pm**, a user sets up a connection to sync data every 24 hours.
* **October 1st, 2:01pm**: sync job runs
* **October 2nd, 2:01pm:** 24 hours have passed since the last sync, so a sync is triggered.
* **October 2nd, 5pm**: The user manually triggers a sync from the UI
* **October 3rd, 2:01pm:** since the last sync was less than 24 hours ago, no sync is run
* **October 3rd, 5:01pm:** It has been more than 24 hours since the last sync, so a sync is run

## Destination namespace

> 
> The location of where a connection replication will store data is referenced as the destination namespace. The destination connectors should create and write records (for both raw and normalized tables) in the specified namespace which should be configurable in the UI thanks to the Namespace Configuration field (or NamespaceDefinition in the API). You can read more about configuring namespaces [here](../namespaces.md).

Reviewer: ```suggestion

The location of where a connection replication will store data is referenced as the destination namespace. The destination connectors should create and write records (for both raw and normalized tables) in the specified namespace which should be configurable in the UI via the Namespace Configuration field (or NamespaceDefinition in the API). You can read more about configuring namespaces [here](../namespaces.md).
```

## Destination stream name

### Prefix stream name

> Stream names refers to table names in a typical RDBMS. But it can also be the name of an API endpoint etc. Similarly to the namespace, stream names can be configured to diverge from their names in the source with a "prefix" field. The prefix is prepended to the source stream name in the destination.

Reviewer 2: ```suggestion
Stream names refer to table names in a typical RDBMS. But it can also be the name of an API endpoint, etc. Similarly to the namespace, stream names can be configured to diverge from their names in the source with a "prefix" field. The prefix is prepended to the source stream name in the destination.
```

> ## Stream by stream customization

Reviewer: ```suggestion
## Stream-specific customization
```

All the customization of namespace and stream names described above will be equally applied to all streams selected for replication in a catalog per connection. If you need more granular customization, stream by stream, for example, or with different logic rules, then you could follow the tutorial on [customizing transformations with dbt](../../tutorials/transformation-and-normalization/transformations-with-dbt.md).

> ## Sync modes

Reviewer: beautiful section

A sync mode governs how Airbyte reads from a source and writes to a destination. Airbyte provides different sync modes to account for various use cases. To minimize confusion, a mode's behavior is reflected in its name. The easiest way to understand Airbyte's sync modes is to understand how the modes are named.

1.  The first part of the name denotes how the source connector reads data from the source:

* Incremental: Read records added to the source since the last sync job. (The first sync using Incremental is equivalent to a Full Refresh)

>   * Method 1: Using a cursor. Generally supported by all sources.

Reviewer: ```suggestion
  * Method 1: Using a cursor. Generally supported by all connectors whose data source allows extracting records incrementally.
```
